calc/src/calc_payout.rs

```diff
@@ -0,0 +1,90 @@
+use crate::debug;
+use core::str::FromStr;
+use log::info;
+use sp_arithmetic::Perbill;
+use wasm_bindgen::prelude::*;
+
+type Balance = u128;
+type RewardPoint = u32;
+
+#[wasm_bindgen]
+#[derive(Debug)]
+pub struct CalcPayout {
+    total_reward_points: RewardPoint,
+    era_payout: Balance,
+}
+
+#[wasm_bindgen]
+impl CalcPayout {
+    pub fn from_params(total_reward_points: u32, era_payout: &str) -> Self {
+        debug::setup();
+
+        info!("from_params({}, {}) -> ", total_reward_points, era_payout);
+
+        let calc_payout = Self {
+            total_reward_points: total_reward_points as RewardPoint,
+            era_payout: Balance::from_str(era_payout).unwrap(),
+        };
+
+        info!("calc_payout: {:#?}", calc_payout);
+
+        calc_payout
+    }
+
+    pub fn calc_payout(
```

Comment:
You could look into making a PR to substrate and so that pallet-staking exposes a function that only does the calculation, and then you simply call into that here, rather than duplicating code. 

Given that it can be done with sensible changes to staking, I will totally support this.